pkg/object/object.go

```diff
@@ -50,16 +71,129 @@ type OpenCompose struct {
 	Volumes  []Volume
 }
 
+func (s *Service) EmptyDirVolumeExists(name string) bool {
+
+	for _, emptyDirVolume := range s.EmptyDirVolumes {
+		if name == emptyDirVolume.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func (o *OpenCompose) VolumeExists(name string) bool {
+	for _, volume := range o.Volumes {
+		if name == volume.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// Documentation about the valid identifiers can be found at
+// https://github.com/kubernetes/community/blob/master/contributors/design-proposals/identifiers.md
+func isValidName(name string) error {
+	if errs := validation.IsDNS1123Subdomain(name); len(errs) != 0 {
+		return fmt.Errorf("%s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func validatePath(path string) error {
+	// TODO: implement me
+	return nil
+}
+
+func (c *Container) Validate() error {
+
+	// validate image name
+	// TODO: implement me
+	// validate Environment
+	// TODO: implement me
+	// validate Ports
+	// TODO: implement me
+
+	// validate Mounts
+	for _, mount := range c.Mounts {
+		// validate volumeName
+		if err := isValidName(mount.VolumeName); err != nil {
+			return err
+		}
+
+		// validate mountPath
+		if err := validatePath(mount.MountPath); err != nil {
+			return err
+		}
+
+		// validate volumeSubPath
+		if err := validatePath(mount.VolumeSubPath); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func (s *Service) Validate() error {
 	// validate service name, like it cannot have underscores, etc.
+	if err := isValidName(s.Name); err != nil {
+		return err
+	}
 
 	// validate containers
+	for _, cnt := range s.Containers {
+		if err := cnt.Validate(); err != nil {
+			return err
+		}
+	}
 
 	// validate replicas
 	if s.Replicas != nil && *s.Replicas < 0 {
 		return fmt.Errorf("Replica count is negative in service: %q", s.Name)
 	}
 
+	// validate emptyDirVolume
+	for _, emptyDir := range s.EmptyDirVolumes {
+		if err := isValidName(emptyDir.Name); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateVolumeMode(volumeMode string) error {
+	switch volumeMode {
+	case "ReadWriteOnce", "ReadOnlyMany", "ReadWriteMany":
+	default:
+		return fmt.Errorf("Invalid Volume AccessMode: %q", volumeMode)
+
+	}
+	return nil
+}
+
+func (v *Volume) Validate() error {
```

Comment:
@surajssd so what do you think about making this private?
I would make all `Validate` function private expect `func (o *OpenCompose) Validate()`. If we are thinking about this being used as library, this is only Validate function that should be exposed.